src/pages/contribute-measurement-page.tsx

```diff
@@ -4,10 +4,18 @@ import { Session } from "@supabase/gotrue-js/src/lib/types";
 import { supabase } from "../utils/supabase-client";
 import { AuthenticatedForm } from "../components/AuthenticatedContributeMeasurementForm";
 import { PasswordlessLogin } from "../components/PasswordlessLogin";
-import { Link } from "react-router-dom/dist/index";
+import { Link, useParams } from "react-router-dom/dist/index";
 
 export const ContributeMeasurementPage: FC = () => {
   const [session, setSession] = useState<Session | null>(null);
+  let suppliedNodeId :string = ""
+  const isNodeId = useParams();
+
+  if (isNodeId.nodeId !== undefined) {
```

Comment:
I think it would be fine to pass `isNodeId.nodeId` directly into `nodeId` into `<AuthenticatedForm>` and change the type so that it's possible to be undefined - the string can still be invalid (empty string) so I think it makes sense to do the validation in the same place (inside `<AuthenticatedForm>`

Reply:
yep, that would be better. Done.